lib/common/ipc_pacemakerd.c

```diff
@@ -62,6 +75,25 @@ pcmk_pacemakerd_api_daemon_state_enum2text(
     return "invalid";
 }
 
+/*!
+ * \internal
+ * \brief Return a friendly string representation of a \p pacemakerd state
+ *
+ * \param[in] state  \p pacemakerd state
+ *
+ * \return A user-friendly string representation of \p state, or
+ *         <tt>"Invalid pacemakerd state"</tt>
+ */
+const char *
+pcmk__pcmkd_state_enum2friendly(enum pcmk_pacemakerd_state state)
+{
+    if ((state >= pcmk_pacemakerd_state_init)
+        && (state <= pcmk_pacemakerd_state_max)) {
+        return pcmkd_state_friendly_str[state];
+    }
+    return "Invalid pacemakerd state";
```

Comment:
usually we just use a switch for this sort of thing rather than an array

Reply:
A switch makes more sense and I'll update. I directly modeled this after `pcmk_pacemakerd_api_daemon_state_enum2text` and didn't think about it. The array prevents redundancy there but doesn't do anything for us here.

At some point I want to deprecate those long names and make them aliases for names like `pcmk_pcmkd_state_enum2text()` or something...